### PR TITLE
Find data page - enhanced contentful fetching #44bvwu

### PR DESCRIPTION
**Pull request by cameronbaney**
# Description

The purpose of this PR is to enhance the contentful endpoint with the following:
- Move from `asyncData()` to a custom method
- Pagination, and offset functionality
- Dynamic search heading based on what is shown

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Visit the [Find Data page](http://localhost:3000/data?type=event)
- Events should load five at a time. On the first page, some events don't content. This is due to how they were entered in Contentful (some are drafts.) I will figure out the best way to do this down the road, but this is a unique situation for the **staging environment** and won't happen on production.
- Use pagination to see more results. Search Heading should update.
- Use the search form in the page's hero to search. Results should update.
- Click on one of the titles. You should be taken to that event's detail page.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by mahamzayed** [comment]
Whenever I search for a keyword in the search bar, the pagination doesn't update accurately. To reproduce:

1. Visit the Find Data page and paginate through the event results as you normally would
2. In the search bar type `blah` and click the search button. The results should refresh to page one as expected. 
3. Click on page 2. The pagination disappears and only shows one page with `No Data`.

File: pages/data/index.vue [hunk 1]
```diff
@@ -165,7 +166,8 @@ export default {
     },
 
     tableData: function() {
-      return compose(pluck('fields'), propOr('', 'items'))(this.searchData)
+      return propOr([], 'items', this.searchData)
+      // return compose(pluck('fields'), propOr('', 'items'))(this.searchData)
```
Comment (mahamzayed) [inline pages/data/index.vue]: Delete if not being used

**Comment by cameronbaney** [comment]
@mahamzayed I have addressed the bug you found.

File: pages/data/index.vue (hunk 1 above)
Comment (cameronbaney) [reply pages/data/index.vue]: Removed.

**Pull request merged**